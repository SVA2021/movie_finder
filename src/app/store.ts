import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
