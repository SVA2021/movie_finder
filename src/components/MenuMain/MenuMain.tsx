import {Home, Menu as MenuIcon, Settings} from '@mui/icons-material';
import {Button, Grid, Popover} from '@mui/material';
import * as React from 'react';
import {FC} from 'react';
import {useNavigate} from "react-router-dom";
import {AccountMenu, SearchMenuMobile, StyledTextItem} from '../../components';

export const MenuMain: FC = () => {

  const navigate = useNavigate();
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);
  const handleOpen = (e: React.MouseEvent<HTMLElement>) => setAnchorEl(e.currentTarget);
  const handleClose = () => setAnchorEl(null);

  function handleClick(path: string): void {
    navigate(path);
    handleClose();
  }

  return (
    <>
      <Button
        id="fade-button"
        aria-haspopup="true"
        aria-controls={open ? 'fade-menu' : undefined}
        aria-expanded={open ? 'true' : undefined}
        onClick={handleOpen}
        sx={{
          backgroundColor: 'transparent',
          color: 'text.secondary',
          boxShadow: 'none',
          "&:hover": {
            color: 'action.hover',
            backgroundColor: 'transparent',
            boxShadow: 'none',
          },
          "&:active": {
            color: 'action.active',
            backgroundColor: 'transparent',
            boxShadow: 'none',
          },
          "&:focus": {
            color: 'common.white',
            backgroundColor: 'transparent',
            boxShadow: 'none',
          },
        }}
      >
        <MenuIcon fontSize='large' />
      </Button>
      <Popover
        data-testid={'main-menu'}
        id="main-menu"
        open={open}
        anchorEl={anchorEl}
        onClose={handleClose}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}
      >
        <Grid container direction={'column'} spacing={2} p={2} bgcolor={'primary.main'}>
          <Grid item xs={12} md={6} lg={3} onClick={() => handleClick('/')} >
            <StyledTextItem ><Home fontSize='medium' />Главная</StyledTextItem>
          </Grid>
          <Grid item xs={12} md={6} lg={3}>
            <StyledTextItem ><Settings fontSize='medium' />Настройки</StyledTextItem>
          </Grid>
          <Grid item xs={12} md={6} lg={3}>
            <AccountMenu />
          </Grid>
          <Grid item xs={12} sx={{display: {xs: 'block', md: 'none', }}}>
            <SearchMenuMobile />
          </Grid>
        </Grid>
      </Popover>
    </>
  );
}