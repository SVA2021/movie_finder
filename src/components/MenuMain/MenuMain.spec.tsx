import {render, screen, fireEvent} from "@testing-library/react";
import {BrowserRouter} from 'react-router-dom';
import {MenuMain} from "./MenuMain";
import {Provider} from 'react-redux';
import {store} from "../../app/store";

describe('render MenuMain', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <MenuMain />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('check button open menu', () => {
    setUp();
    fireEvent.click(screen.getByRole('button'))
    expect(screen.getByTestId('main-menu')).toBeInTheDocument();
  });

})