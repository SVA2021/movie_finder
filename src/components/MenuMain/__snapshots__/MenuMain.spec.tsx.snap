// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MenuMain snapshot test 1`] = `
<button
  aria-haspopup="true"
  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1qbbts5-MuiButtonBase-root-MuiButton-root"
  id="fade-button"
  tabindex="0"
  type="button"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
    data-testid="MenuIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
